src/Core/FlatpakBackend.vala

```diff
@@ -1690,6 +1693,64 @@ public class AppCenterCore.FlatpakBackend : Backend, Object {
         return package_list[id];
     }
 
+    private void repair_internal (Job job) {
+        unowned var args = (RepairArgs)job.args;
+        unowned var cancellable = args.cancellable;
+
+        bool success = true;
+
+        try {
+            int status;
+
+            Process.spawn_command_line_sync ("flatpak --user repair", null, null, out status);
+
+            if (status != 0) {
+                success = false;
+            }
+        } catch (Error e) {
+            job.error = e;
+            job.results_ready ();
+            return;
+        }
+
+        if (!success || cancellable.is_cancelled ()) {
+            job.result = Value (typeof (bool));
+            job.result = success;
+            job.results_ready ();
+        }
+
+        try {
+            int status;
+
+            Process.spawn_command_line_sync ("pkexec flatpak --system repair", null, null, out status);
+
+            if (status != 0) {
+                success = false;
+            }
+        } catch (Error e) {
+            job.error = e;
+            job.results_ready ();
+            return;
+        }
+
+        job.result = Value (typeof (bool));
+        job.result = success;
+        job.results_ready ();
+    }
+
+    public async bool repair (Cancellable? cancellable = null) throws GLib.Error {
+        var job_args = new RepairArgs ();
+        job_args.cancellable = cancellable;
+
+        var job = yield launch_job (Job.Type.REPAIR, job_args);
+        if (job.error != null) {
+            warning ("Could not repair user Flatpak: %s", job.error.message);
```

Comment:
We should probably not silently fail and make sure we throw an error dialog if we're unable to repair